src/OneExplorer/OneStorage.ts

```diff
@@ -0,0 +1,169 @@
+   * A map of ConfigObj (key: cfg path)
+   */
+  private _cfgToCfgObjMap: ConfigObjMap;
+  /**
+   * A map of BaseModel path to Cfg path
+   */
+  private _baseModelToCfgsMap: StringMap;
+
+  /**
+   * Get the list of .cfg files within the workspace
+   * @param root  the file or directory,
+   *              which MUST exist in the file system
+   */
+  private _getCfgList(root: string = obtainWorkspaceRoot()): string[] {
+    /**
+     * Returns every file inside directory
+     * @todo Check soft link
+     * @param root
+     * @returns
+     */
+    const readdirSyncRecursive = (root: string): string[] => {
+      if (fs.statSync(root).isFile()) {
+        return [root];
+      }
+
+      let children: string[] = [];
+      if (fs.statSync(root).isDirectory()) {
+        fs.readdirSync(root).forEach(val => {
+          children = children.concat(readdirSyncRecursive(path.join(root, val)));
+        });
+      }
+      return children;
+    };
+
+    try {
+      fs.statSync(root);
+    } catch {
+      Logger.error('OneExplorer', 'getCfgList', 'called on not existing directory or file.');
+      return [];
+    }
+
+    return readdirSyncRecursive(root).filter(val => val.endsWith('.cfg'));
+  }
+
+  private _getCfgToCfgObjMap(cfgList: string[]): ConfigObjMap {
+    let map: ConfigObjMap = {};
+
+    cfgList.forEach(cfg => {
+      map[cfg] = ConfigObj.createConfigObj(vscode.Uri.file(cfg));
+    });
+
+    return map;
+  }
+
+  private _getBaseModelToCfgsMap(cfgList: string[], cfgToCfgObjMap: ConfigObjMap): StringMap {
+    let map: StringMap = {};
+
+    cfgList.forEach(cfg => {
+      const cfgObj = cfgToCfgObjMap[cfg];
+      if (cfgObj) {
+        cfgObj.getBaseModelsExists.forEach(baseModelArtifact => {
+          if (!map[baseModelArtifact.path]) {
+            map[baseModelArtifact.path] = [];
+          }
+
+          if (!map[baseModelArtifact.path].includes(cfg)) {
+            map[baseModelArtifact.path].push(cfg);
+          }
+        });
+      }
+    });
+
+    return map;
+  }
+
+  private constructor() {
+    this._cfgList = this._getCfgList();
+    this._cfgToCfgObjMap = this._getCfgToCfgObjMap(this._cfgList);
+    this._baseModelToCfgsMap = this._getBaseModelToCfgsMap(this._cfgList, this._cfgToCfgObjMap);
+  }
+
+  private static _obj: OneStorage|undefined;
+
+  /**
+   * Get cfg lists which refers the base model path
+   * @param baseModelPath
+   * @return a list of cfg path or 'undefined'
+   */
+  public static getCfgs(baseModelPath: string): string[]|undefined {
+    return OneStorage.get()._baseModelToCfgsMap[baseModelPath];
+  }
+
+  /**
+   * Get cfgObj from the map
+   */
+  public static getCfgObj(cfgPath: string): ConfigObj|null {
+    return OneStorage.get()._cfgToCfgObjMap[cfgPath];
+  }
+
+  /**
+   * Get a singleton object
+   */
+  private static get(): OneStorage {
+    if (!OneStorage._obj) {
+      OneStorage._obj = new OneStorage;
+    }
+    return OneStorage._obj;
+  }
+
+  static reset(): void {
```

Comment:
Just for consistency? :D

```suggestion
  public static reset(): void {
```